src/devices/machine/mc68328.h

```diff
@@ -124,10 +124,15 @@ class mc68328_base_device : public m68000_device
 
 	DECLARE_WRITE_LINE_MEMBER(irq5_w);
 
-	template <typename T>
-	std::enable_if_t<lcd_info_changed_delegate::supports_callback<T>::value> set_lcd_info_changed(T &&callback, const char *name)
+	template <typename F>
+	std::enable_if_t<lcd_info_changed_delegate::supports_callback<F>::value> set_lcd_info_changed(F &&callback, const char *name)
 	{
-		m_lcd_info_changed_cb.set(std::forward<T>(callback), name);
+		m_lcd_info_changed_cb.set(std::forward<F>(callback), name);
+	}
+	template <typename T, typename F>
+	std::enable_if_t<lcd_info_changed_delegate::supports_callback<F>::value> set_lcd_info_changed(T &&target, F &&callback, const char *name)
+	{
+		m_lcd_info_changed_cb.set(std::forward<T>(target), std::forward<F>(callback), name);
 	}
```

Comment:
Do you actually need the two overloads?  The only reason `screen_device` does that kind of SFINAE trickery is because it has two different signatures for screen update functions and uses different delegates.  Could you do this with a single setter as:
```
	template <typename... T>
	void set_lcd_info_changed(T &&... args)
	{
		m_lcd_info_changed_cb.set(std::forward<T>(args)...);
	}
```

Reply:
I'll give that a shot, thanks for the suggestion.

Reply:
Yep, this suggestion works great. Thanks!